tests/unit/dogstatsd/test_statsd.py

```diff
@@ -402,6 +403,22 @@ def test_accessing_socket_multiple_times_returns_same_socket(self):
         t.assert_equal(fresh_socket, dogpound.get_socket())
         t.assert_not_equal(FakeSocket(), dogpound.get_socket())
 
+    def test_tags_from_environment(self):
+        os.environ['DOGSTATSD_TAGS'] = 'country:china,age:45,blue'
+        statsd = DogStatsd()
+        del(os.environ['DOGSTATSD_TAGS'])
+        statsd.socket = FakeSocket()
+        statsd.gauge('gt', 123.4)
+        t.assert_equal('gt:123.4|g|#country:china,age:45,blue', statsd.socket.recv())
+
+    def test_tags_from_environment_and_constant(self):
+        os.environ['DOGSTATSD_TAGS'] = 'country:china,age:45,blue'
+        statsd = DogStatsd(constant_tags=['country:canada', 'red'])
+        del(os.environ['DOGSTATSD_TAGS'])
+        statsd.socket = FakeSocket()
+        statsd.gauge('gt', 123.4)
+        t.assert_equal('gt:123.4|g|#country:canada,red,country:china,age:45,blue', statsd.socket.recv())
+
```

Comment:
Thanks for the additional tests @ewdurbin !

Instead of potentially overriding, removing the host's `DOGSTATSD_TAGS ` environment variable, can you use the [`preserve_environ_datadog `](https://github.com/DataDog/datadogpy/blob/a60d519f774b28151339ffb20bd7f88905f12d0e/tests/unit/api/test_api.py#L32) decorator please ? We could also consider moving the method to a new `tests/util/` module :smile: .